drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -299,6 +400,23 @@ local function assign_child_profile(device, child_ep)
   return profile or "switch-binary"
 end
 
+local function do_configure(driver, device)
+  local energy_eps = device:get_endpoints(clusters.ElectricalEnergyMeasurement.ID)
+  local power_eps = device:get_endpoints(clusters.ElectricalPowerMeasurement.ID)
+  local profile_name = nil
+  if #energy_eps > 0 and #power_eps > 0 then
+    profile_name = "power-energy-powerConsumption"
+  elseif #energy_eps > 0 then
+    profile_name = "electrical-energy-powerConsumption"
+  elseif #power_eps > 0 then
+    profile_name = "electrical-power"
+  end
+
+  if profile_name then
+    device:try_update_metadata({ profile = profile_name })
+  end
+end
```

Comment:
@nickolas-deboom like I'd mentioned, perhaps we can add some handling to do_configure? Maybe we can implement something here like we do in thermostat where we check per device type, so as not to break anything that's already made?

Reply:
Since the other device types don't need this kind of functionality